app/models/recipe.rb

```diff
@@ -1,3 +1,9 @@
 class Recipe < ApplicationRecord
    validates_presence_of :name, :complexity, :genre
+   has_many :recipe_ingredients
+   has_many :ingredients, through: :recipe_ingredients
+
+   def total_ingredient_cost
+      ingredients.sum(:cost)
+   end
```

Comment:
Great AR, and making use of your association in your `total_ingredient_cost` method!